src/lib/helpers.ts

```diff
@@ -16,3 +17,53 @@ export const blockConfig = (
   startBlock: Math.min(Math.max(start || 0, startBlock), end || Number.MAX_SAFE_INTEGER),
   endBlock: end,
 });
+
+// default rate limit for request per second to RPC endpoints
+const DEFAULT_RPC_RATE_LIMIT = 50;
+
+/**
+ * Reads the RPC request rate limit for a given chain ID from the environment
```

Comment:
```suggestion
 * Gets the RPC request rate limit for a given chain ID.
```

Then suggest moving or removing the remainder of this first paragraph.

The comments here are for the overall function. Generally it's best not to document the implementation details of the function here. These comments on the outside of the function should just be for the callers of this function and those callers generally shouldn't be concerned with purely internal implementation details. Ex: that the value is read from environment variables.

You're welcome to move this paragraph of comments to the first line of code inside the function. These details are specific to the internal implementation.

The other paragraphs here are great to keep.